gmso/tests/test_mcf.py

```diff
@@ -149,3 +157,78 @@ def test_scaling_factors(self, n_typed_ar_system):
         assert np.allclose(float(mcf_data[-4][1]), 0.4)
         assert np.allclose(float(mcf_data[-4][2]), 0.6)
         assert np.allclose(float(mcf_data[-4][3]), 1.0)
+
```

Comment:
I would add a test for if the atomtype names get cut off or something. Otherwise, I think you're just missing 3-5 tests that errors get raised properly if something is done wrong. I think they should include:
1. Try to write out a non-parameterized topology
2. Try to write out something with an incompatible potential.
3. Try to write out something with too many atomtypes
4. Try to write out something with two nonbonded potentials
5. Write out something that isn't charge neutral
6. Write out something that is multiple molecules
- This one maybe wouldn't raise an error if we design the writer to just make that write out two separate files.

We also need a test for the ring structures since those are handled oddly;.

Reply:
Thanks Cal. I added your suggestions to my todo list.